crates/operator/src/lib.rs

```diff
@@ -234,6 +234,17 @@ impl Operator {
             .await
             .map_err(|_| OperatorError::OperatorIdError)?;
 
+        // Check that the operator ID from the BLS key pair is the same as the operator ID from the contract
```

Comment:
Let's document this check in the method documentation, like:

```rust
/// It also performs some sanity checks:
/// - Verifies the operator ID derived from the BLS keypair is equal to the operator ID registered in the contracts.
```

Also, the documentation wrongly says "This method does not register the operator"